### PR TITLE
Import level two licence sectors into Specialist Publisher

### PR DESCRIPTION
**Pull request by leenagupte**
Trello: https://trello.com/c/TiYLM2hE

# What's changed and why?

It has been agreed with Content Designers and User Researchers to tag migrated licences in Specialist Publisher to only level two sectors (from Licence Finder).

The hierarchical structure for sectors (3 levels) is being dropped and licence documents will just be tagged to the 2nd level. This is due to limitations to support hierarchical tagging in Specialist Publisher / Finder Frontend and to mitigate bespoke work just for User Research.

Activities do not need to be imported as activities don't make complete sense. They mostly map 1-2-1 with individual licences but have different names to the licences.

Country tagging will remain and is already configured in Specialist Publisher.

The tagging of licence documents to the level sectors will be accomplished in a future data migration.

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

**Review by 1pretz1 — APPROVED** [review]
Looks good!

File: app/services/licence_transaction/import_industry_sectors.rb [hunk 1]
```diff
@@ -4,6 +4,10 @@ def self.call
   end
```
Comment (1pretz1) [inline app/services/licence_transaction/import_industry_sectors.rb]: Minor: we should use encapsulation for the private methods

File: spec/services/licence_transaction/import_industry_sectors_spec.rb [hunk 2]
```diff
@@ -1,21 +1,21 @@
 require "spec_helper"
 
 RSpec.describe LicenceTransaction::ImportIndustrySectors do
```
Comment (1pretz1) [inline spec/services/licence_transaction/import_industry_sectors_spec.rb]: I think we're now testing the wrong method (should be `call`) and as a result we're missing some tests for the features that have been added (reading / writing to JSON file). If we find it useful to have `imported_json_data` as its own public method then we should probably test both.

**Review by 1pretz1 — COMMENTED** [review]
Changes look good, nice automation! I've just made a couple of comments. We could also create a rake task for this import which is a bit nicer than having to run commands in a console.

File: spec/services/licence_transaction/import_industry_sectors_spec.rb (hunk 2 above)
Comment (leenagupte) [reply spec/services/licence_transaction/import_industry_sectors_spec.rb]: You caught me! I'll have a bit of refactor and find a nice way to test the JSON file.

**Comment by leenagupte** [comment]
@1pretz1 Thanks for your review. I've addressed your comments and fixed up.

File: app/services/licence_transaction/import_industry_sectors.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
+class LicenceTransaction::ImportIndustrySectors
```
Comment (1pretz1) [inline app/services/licence_transaction/import_industry_sectors.rb]: This feels like it should live in lib as it's only used by a single rake task and not part of the main application code

**Review by 1pretz1 — APPROVED** [review]
Nice work! Approving as my only comment is minor.

File: app/services/licence_transaction/import_industry_sectors.rb (hunk 3 above)
Comment (leenagupte) [reply app/services/licence_transaction/import_industry_sectors.rb]: Hmmm, I thought about this, and figured it was really an import service with the potential to be run a few times. Maybe even refactored to be used in the scripts to update the sector names and re-tag licences. 

The name is probably wrong though. I think it should have been named "IndustrySectorsImporter"

Comment (1pretz1) [reply app/services/licence_transaction/import_industry_sectors.rb]: I still think it should live in `lib` unless it is being used by the application at runtime as everything else in the services directory is related to the main application code. Interestingly, we do have an `importers` directory in `lib`: https://github.com/alphagov/specialist-publisher/tree/589ad0fee62548e86e979aec74841044e841f959/lib.

Agreed on the proposed name 👍

Comment (leenagupte) [reply app/services/licence_transaction/import_industry_sectors.rb]: Agreed on name and location. I'll move the class and test files into lib. I noticed the same "importer" folder when I was reviewing your PR! https://github.com/alphagov/specialist-publisher/pull/2210#discussion_r1072103633

**Comment by leenagupte** [comment]
@1pretz1 I've renamed the files and moved everything to under "lib" as agreed. The changes are ready for a re-review.

**Review by 1pretz1 — APPROVED** [review]
Nice!

**Pull request merged**